### ISSUE TITLE
Document how to run Docker container with custom arguments

### ISSUE THREAD

**Issue by gajus** [joelgriffith/browserless#35]
Having read README.md, it is unclear how to start a docker container with customer arguments.

I have tried:

```bash
$ docker run -p 3000:3000 browserless/chrome --proxy-server=127.0.0.1:8050

```

But that does not work.

Furthermore, even if this did work, how would one then acquire the host IP address? That requires checking ifconfig on the container.

Perhaps whats needed is an initialisation script (JavaScript) that could be loaded to the container and used to configure the instance.

**Comment by joelgriffith** [comment]
Hey there, I’m a tad unclear about the proxy-server (since we don’t have docs on that), which could be making matters worse. There are more elaborate docs here: https://docs.browserless.io/docs/docker.html. 

Let me know if that doesn’t cover what you’re trying to achieve

**Comment by gajus** [comment]
> Let me know if that doesn’t cover what you’re trying to achieve

`--proxy-server` arg should be passed to Chrome when launching it, e.g.

https://github.com/GoogleChrome/puppeteer/issues/336

Here the docs talk about passing proxy-server parameter via the query parameters, e.g.

https://docs.browserless.io/docs/chrome.html

How does this work? Does Browserless container start a new Chrome instance when new connection is made?

**Comment by joelgriffith** [comment]
> How does this work? Does Browserless container start a new Chrome instance every time a new connection is made?

Yup, that's how it functions by default. You can specify it to re-use Chrome instances, but that work is still early in its life-cycle.

**Comment by TAnas0** [comment]
@gajus The link is dead and I stumbled on this issue trying to specify the Websocket port on which Puppeteer listens.

To my understanding, its default value is 0 and means that it will be set randomly high, and I am getting errors about connection being refused to endpoints : `127.0.0.1:<port>` where `port` is random.

If this rings any bells, or if you have an alternative link that would be awesome.

Thanks
